File: test_app/integration_tests/android_mocks/src/main/java/com/voxeet/android/media/spatialisation/SpatialDirection.java
package com.voxeet.android.media.spatialisation;

public class SpatialDirection {
    /**
     * The Euler rotation about the x-axis, in degrees.
     */
    public final double x;
    /**
     * The Euler rotation about the y-axis, in degrees.
     */
    public final double y;
    /**
     * The Euler rotation about the z-axis, in degrees.
     */
    public final double z;

    public SpatialDirection(double x,
                            double y,
                            double z) {
        this.x = x;
        this.y = y;
        this.z = z;
    }
}

File: dolbyio_comms_sdk_flutter/android/src/main/kotlin/io/dolby/comms/sdk/flutter/mapper/SpatialDirection.kt
package io.dolby.comms.sdk.flutter.mapper

import com.voxeet.android.media.spatialisation.SpatialDirection

class SpatialDirectionMapper {
    companion object {
        fun fromMap(map: Map<String, Any?>?): SpatialDirection? {
            if (map == null || !map.contains("x") || !map.contains("y") || !map.contains("z"))
                return null
            return SpatialDirection(map["x"] as Double, map["y"] as Double, map["z"] as Double)
        }
    }
}
